src/Util/GeometryUtil/GeometryUtil.js

```diff
@@ -0,0 +1,292 @@
+    // reprojecting them to the internal turf.js projection 'EPSG:4326'.
+    const turfPolygon = geoJsonFormat.writeFeatureObject(polygonFeat);
+    const turfLine = geoJsonFormat.writeFeatureObject(lineFeat);
+
+    // Union both geometries to a set of MultiLineStrings.
+    const unionGeom = union(polygonToLine(turfPolygon), turfLine);
+
+    // Buffer the input polygon to take great circle (un-)precision
+    // into account.
+    const bufferedTurfPolygon = buffer(turfPolygon, tolerance, {
+      units: 'meters'
+    });
+
+    let filteredSegments = [];
+
+    // Iterate over each segment and remove any segment that is not covered by
+    // the (buffered) input polygon.
+    segmentEach(unionGeom, (currentSegment, featureIndex, multiFeatureIndex) => {
+      const segmentCenter = centroid(currentSegment);
+      const isSegmentInPolygon = booleanPointInPolygon(segmentCenter, bufferedTurfPolygon);
+
+      if (isSegmentInPolygon) {
+        if (!filteredSegments[multiFeatureIndex]) {
+          filteredSegments[multiFeatureIndex] = [];
+        }
+
+        if (filteredSegments[multiFeatureIndex].length === 0) {
+          filteredSegments[multiFeatureIndex].push(
+            getCoords(currentSegment)[0],
+            getCoords(currentSegment)[1]
+          );
+        } else {
+          filteredSegments[multiFeatureIndex].push(
+            getCoords(currentSegment)[1]
+          );
+        }
+      }
+    });
+
+    // Rebuild the unioned geometry based in the filtered segments.
+    const filteredUnionGeom = multiLineString(filteredSegments);
+
+    // Polygonize the lines.
+    const polygonizedUnionGeom = polygonize(filteredUnionGeom);
+
+    // Return as Array of ol.Features.
+    const features = geoJsonFormat.readFeatures(polygonizedUnionGeom);
+    if (polygon instanceof OlFeature && line instanceof OlFeature) {
+      return features;
+    } else {
+      return features.map(f => f.getGeometry());
+    }
+  }
+
+  /**
+   * Adds a buffer to a given geometry.
+   *
+   * @param {ol.geom.Geometry | ol.Feature} geometry The geometry.
+   * @param {Integer} buffer The buffer to add in meters.
+   * @param {String} projection A projection as EPSG code.
+   *
+   * @returns {ol.geom.Geometry | ol.Feature} The geometry or feature with the added buffer.
+   */
+  static addBuffer (geometry, radius = 0, projection = 'EPSG:3857') {
+    if (radius === 0) {
+      return geometry;
+    }
+    const geoJsonFormat = new OlFormatGeoJSON({
+      dataProjection: 'EPSG:4326',
+      featureProjection: projection
+    });
+    const geoJson = geometry instanceof OlFeature
+      ? geoJsonFormat.writeFeatureObject(geometry)
+      : geoJsonFormat.writeGeometryObject(geometry);
+    const buffered = buffer(geoJson, radius, {
+      units: 'meters'
+    });
+    if (geometry instanceof OlFeature) {
+      return geoJsonFormat.readFeature(buffered);
+    } else {
+      return geoJsonFormat.readGeometry(buffered.geometry);
```

Comment:
I *think* I like one return type better. but this is not blocking this.

Reply:
Docs are enhanced to clearify why we two different return types.